packages/compiler/src/compiler/compile.test.ts

```diff
@@ -304,7 +304,7 @@ describe('message contents', async () => {
 })
 
 describe('reference tags', async () => {
-  it('can reference a prompt', async () => {
+  it('always fails', async () => {
```

Comment:
Compiler now cannot resolve references, it will always fail with a custom error message. Instead, now it is required to resolve all references before compiling.

All tests about references in compiling have been removed.